XO.py

```diff
@@ -0,0 +1,157 @@
+from random import shuffle
+
+# size of board
+N = 3
+
+# Utility functions
+
+
+def print_board(board):
+    print()
+    print('Board:')
+    for line_ in board:
+        for item in line_:
+            print(item, end='')
+        print()
+    print()
+
+
+def check_input_validity(input_type, curr_player):
+    not_valid = True
+    while not_valid:
+        print('{}, please insert the {} number'.format(curr_player, input_type))
+        try:
+            input_value = int(input())
+        except:
+            print('{}, please insert a valid {} number'.format(curr_player, input_type))
+            continue
+        if input_value < 0 or input_value >= N:
+            print('{}, please insert a valid {} number'.format(curr_player, input_type))
+            continue
+        not_valid = False
+    return input_value
+
+
+def check_line(board, line, curr_player, curr_player_symbol, is_win):
+    for sym in board[line]:
+        if sym != curr_player_symbol:
+            is_win = False
+            break
+    if is_win:
+        return declare_win(curr_player)
+
+
+def check_col(board, col, curr_player, curr_player_symbol, is_win):
+    for line in board:
+        if line[col] != curr_player_symbol:
+            is_win = False
+            break
+    if is_win:
+        return declare_win(curr_player)
+
+
+def check_main_diagonal(board, line, col, curr_player, curr_player_symbol, is_win):
+    if line == col:
+        for i in range(N):
+            if board[i][i] != curr_player_symbol:
+                is_win = False
+                break
+    else:
+        is_win = False
+
+    if is_win:
+        return declare_win(curr_player)
+
+
+def check_opposite_diagonal(board, line, col, curr_player, curr_player_symbol, is_win):
+    if line + col == N - 1:
+        j = 2
+        for i in range(N):
+            if j < 0:
+                is_win = False
+                break
+            if board[i][j] != curr_player_symbol:
+                is_win = False
+                break
+            j-=1
+    else:
+        is_win = False
+
+    if is_win:
+        return declare_win(curr_player)
+
+
+def declare_win(curr_player):
+    print('The winner is {}!!!'.format(curr_player))
+    print_board(board)
+    return(True)
+
+
+def switch_users(curr_player_, x_player, o_player):
+    if curr_player_ == x_player:
+        curr_player_ = o_player
+        curr_player_sym = 'o'
+    else:
+        curr_player_ = x_player
+        curr_player_sym = 'x'
+    return curr_player_, curr_player_sym
+
+
+# initiate board

```

Comment:
Better to put all code inside functions - (`initiate_board`, `main`, etc...)

Reply:
Done